app/workers/functions.php

```diff
@@ -463,25 +436,36 @@ public function execute(string $trigger, string $projectId, string $executionId,
         $stderr = '';
 
         $executionStart = \microtime(true);
-        
-        $exitCode = Console::execute("docker exec ".\implode(" ", $vars)." {$container} {$command}"
-            , '', $stdout, $stderr, $function->getAttribute('timeout', (int) App::getEnv('_APP_FUNCTIONS_TIMEOUT', 900)));
+
+        $exitCode = 0;
+
+        try {
+            $exitCode = (int)!$orchestration->execute(
+                name: $container, 
+                command: $orchestration->parseCommandString($command), 
+                stdout: $stdout, 
+                stderr: $stderr, 
+                vars: $vars, 
+                timeout: $function->getAttribute('timeout', (int) App::getEnv('_APP_FUNCTIONS_TIMEOUT', 900)));
+        } catch (TimeoutException $e) {
+            $exitCode = 124;
+        }
 
         $executionEnd = \microtime(true);
         $executionTime = ($executionEnd - $executionStart);
         $functionStatus = ($exitCode === 0) ? 'completed' : 'failed';
 
-        Console::info("Function executed in " . ($executionEnd - $executionStart) . " seconds with exit code {$exitCode}");
+        Console::info('Function executed in ' . ($executionEnd - $executionStart) . ' seconds, status: ' . $functionStatus);
 
         Authorization::disable();
         
         $execution = $database->updateDocument(array_merge($execution->getArrayCopy(), [
             'tagId' => $tag->getId(),
             'status' => $functionStatus,
-            'exitCode' => $exitCode,
+            'exitCode' => ($exitCode === 0 ? 0 : 1),
```

Comment:
Why not just throw the exit code as-is? As written, we lose the context as to why it may fail (124 timeout)